sql/core/src/test/scala/org/apache/spark/sql/streaming/EventTimeWatermarkSuite.scala

```diff
@@ -132,13 +132,13 @@ class EventTimeWatermarkSuite extends StreamTest with BeforeAndAfter with Loggin
       CheckLastBatch(),
       AddData(inputData, 25),   // Advance watermark to 15 seconds
       CheckLastBatch(),
-      assertNumStateRows(3),
+      assertNumStateRows(3, 1),
       AddData(inputData, 25),   // Emit items less than watermark and drop their state
       CheckLastBatch((10, 5)),
-      assertNumStateRows(2),
+      assertNumStateRows(2, 1),
       AddData(inputData, 10),   // Should not emit anything as data less than watermark
       CheckLastBatch(),
-      assertNumStateRows(2)
+      assertNumStateRows(2, 1)  // We also processed the data 10, which is less than watermark
```

Comment:
note this is `assertNumStateRows(2, 1)` without this patch, and is `assertNumStateRows(2, 0)` with this patch